internal/engine/pod.go

```diff
@@ -151,71 +130,96 @@ func ensureManifestTargetWithPod(state *store.EngineState, pod *v1.Pod) (*store.
 }
 
 // Fill in container fields on the pod state.
-func populateContainerStatus(ctx context.Context, manifest model.Manifest, podInfo *store.Pod, pod *v1.Pod, cStatus v1.ContainerStatus) {
-	cName := k8s.ContainerNameFromContainerStatus(cStatus)
-	podInfo.ContainerName = cName
-	podInfo.ContainerReady = cStatus.Ready
-
-	cID, err := k8s.ContainerIDFromContainerStatus(cStatus)
+func populateContainers(ctx context.Context, manifest model.Manifest, podInfo *store.Pod, pod *v1.Pod) {
+	blessedCID, err := getBlessedContainerID(manifest.ImageTargets, pod)
 	if err != nil {
-		logger.Get(ctx).Debugf("Error parsing container ID: %v", err)
+		logger.Get(ctx).Debugf("Populating containers for pod: %v", err)
 		return
 	}
-	podInfo.ContainerID = cID
 
-	cRef, err := container.ParseNamed(cStatus.Image)
-	if err != nil {
-		logger.Get(ctx).Debugf("Error parsing container image ID: %v", err)
-		return
-	}
-	podInfo.ContainerImageRef = cRef
-
-	ports := make([]int32, 0)
-	cSpec := k8s.ContainerSpecOf(pod, cStatus)
-	for _, cPort := range cSpec.Ports {
-		ports = append(ports, cPort.ContainerPort)
-	}
-	podInfo.ContainerPorts = ports
-
-	forwards := PopulatePortForwards(manifest, *podInfo)
-	if len(forwards) < len(manifest.K8sTarget().PortForwards) {
-		logger.Get(ctx).Infof(
-			"WARNING: Resource %s is using port forwards, but no container ports on pod %s",
-			manifest.Name, podInfo.PodID)
-	}
+	// Clear containers on the pod
+	podInfo.Containers = nil
 
-	// HACK(maia): Go through ALL containers (except tilt-synclet), grab minimum info we need
-	// to stream logs from them.
-	var cInfos []store.ContainerInfo
-	for _, cStat := range pod.Status.ContainerStatuses {
-		if cStat.Name == sidecar.SyncletContainerName {
-			// We don't want logs for the Tilt synclet.
+	for _, cStatus := range pod.Status.ContainerStatuses {
+		if cStatus.Name == sidecar.SyncletContainerName {
+			// We don't want logs, status, etc. for the Tilt synclet.
 			continue
 		}
 
-		cID, err := k8s.ContainerIDFromContainerStatus(cStat)
+		cName := k8s.ContainerNameFromContainerStatus(cStatus)
+
+		cID, err := k8s.ContainerIDFromContainerStatus(cStatus)
 		if err != nil {
 			logger.Get(ctx).Debugf("Error parsing container ID: %v", err)
 			return
```

Comment:
should this be a `continue` rather than a `return`? 

alternatively, it might be worth moving this to a helper function so that it's harder to make this mistake